### PR TITLE
Memalloy ptx

### PR DESCRIPTION
**Pull request by tonghaining**
Add more tests from memalloy

### CODE REVIEW

**Comment by hernanponcedeleon** [comment]
How should we match the format from memalloy, e.g.
```
//IRIW (1_unique)
//(T0 CTA) (T1,T2,T3 CTA)
//(T1 membar.cta) (T3 membar.gl)
irreflexive [
             ((X.rf) & intra_gl_raw[X]) .
             (((X.R) -> (X.R)) & cta_fence_raw[X] - X.sloc) .
             ((fr[X]) & intra_cta_raw[X]) .
             ((X.rf) & intra_cta_raw[X]) .
             (((X.R) -> (X.R)) & gl_fence_raw[X] - X.sloc) .
             ((fr[X]) & intra_gl_raw[X])
            ]
```
to litmus tests?

@tonghaining can you explain please?

**Comment by tonghaining** [comment]
I used the device tests as reference like the one under [GTX 540](https://multicore.doc.ic.ac.uk/gpu-litmus/entries/gtx540-IRIW+membar.cta+membar.gl-p0::p1:p2:p3-xgyg.txt)

**Comment by hernanponcedeleon** [comment]
> I used the device tests as reference like the one under [GTX 540](https://multicore.doc.ic.ac.uk/gpu-litmus/entries/gtx540-IRIW+membar.cta+membar.gl-p0::p1:p2:p3-xgyg.txt)

Then it makes more sense to put the link to this in our litmus since the information in the current links is just too cryptic

File: dartagnan/src/test/resources/PTXv7_5-expected.csv [hunk 1]
```diff
@@ -207,9 +207,29 @@ litmus/PTX/Manual/PC-bar-sync-sync-1.litmus,1
...
+litmus/PTX/Memalloy/IRRWIW4.litmus,1
 litmus/PTX/Memalloy/IRRWIW_uniproc.litmus,1
+litmus/PTX/Memalloy/IRWIW1.litmus,0
+litmus/PTX/Memalloy/IRWIW2.litmus,0
 litmus/PTX/Memalloy/IRWIW_gl_cta.litmus,1
+litmus/PTX/Memalloy/RWC1.litmus,1
+litmus/PTX/Memalloy/RWC2.litmus,1
 litmus/PTX/Memalloy/RWC_uniproc.litmus,1
+litmus/PTX/Memalloy/WRC1.litmus,1
+litmus/PTX/Memalloy/WRC2.litmus,1
```
Comment (hernanponcedeleon) [inline dartagnan/src/test/resources/PTXv7_5-expected.csv:226]: All tests should also be added to `PTX6_0-exspected.csv`

File: litmus/PTX/Memalloy/IRWIW1.litmus [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 0,gpu 0           | P2@cta 0,gpu 0           | P3@cta 1,gpu 0           ;
+ ld r0, 2                 | ld.weak r0, x            | ld r0, 2                 | ld.weak r0, x            ;
+ st.weak x, r0            | fence.sc.cta             | st.weak x, r0            | fence.sc.gpu             ;
+                          | ld r2, 1                 |                          | ld r2, 1                 ;
+                          | st.weak y, r2            |                          | st.weak y, r2            ;
+exists
+(x == 0 /\ y == 2 /\ P1:r0 == 2 /\ P3:r0 == 2)
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/IRWIW1.litmus:21]: it should be `x == 2`

File: litmus/PTX/Memalloy/IRWIW2.litmus [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           | P3@cta 1,gpu 0           ;
+ ld r0, 2                 | ld.weak r0, x            | ld r0, 2                 | ld.weak r0, x            ;
+ st.weak x, r0            | fence.sc.cta             | st.weak x, r0            | fence.sc.gpu             ;
+                          | ld r2, 1                 |                          | ld r2, 1                 ;
+                          | st.weak y, r2            |                          | st.weak y, r2            ;
+exists
+(x == 0 /\ y == 2 /\ P1:r0 == 2 /\ P3:r0 == 2)
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/IRWIW2.litmus:21]: it should be `x == 2`

File: litmus/PTX/Memalloy/RWC1.litmus [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+P1:r2=0;
+P2:r0=0;
+P2:r2=0;
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           ;
+ ld r0, 1                 | ld.weak r0, x            | ld r0, 1                 ;
+ st.weak x, r0            | fence.sc.cta             | st.weak x, r0            ;
+                          | ld.weak r2, y            | fence.sc.sys             ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/RWC1.litmus:17]: shouldn't it be `fence.sc.gpu`?

File: litmus/PTX/Memalloy/RWC2.litmus [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+P1:r0=0;
+P1:r2=0;
+P2:r0=0;
+P2:r2=0;
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           ;
+ ld r0, 1                 | ld.weak r0, x            | ld r0, 1                 ;
+ st.weak x, r0            | fence.sc.sys             | st.weak x, r0            ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/RWC2.litmus:16]: should it be `fence.sc.g[u`?

File: litmus/PTX/Memalloy/WRC1.litmus [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+P1:r0=0;
+P1:r2=0;
+P2:r0=0;
+P2:r2=0;
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           ;
+ ld r0, 1                 | ld.weak r0, x            | ld.weak r0, x            ;
+ st.weak x, r0            | fence.sc.cta             | fence.sc.sys             ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/WRC1.litmus:16]: sys -> gpu

File: litmus/PTX/Memalloy/WRC2.litmus [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+P1:r0=0;
+P1:r2=0;
+P2:r0=0;
+P2:r2=0;
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           ;
+ ld r0, 1                 | ld.weak r0, x            | ld.weak r0, x            ;
+ st.weak x, r0            | fence.sc.sys             | fence.sc.cta             ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/WRC2.litmus:16]: sys -> gpu

File: litmus/PTX/Memalloy/WRR+2W1.litmus [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+P0:r0=0;
+P1:r0=0;
+P1:r2=0;
+P2:r0=0;
+P2:r2=0;
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           ;
+ ld r0, 1                 | ld.weak r0, x            | ld r0, 2                 ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/WRR+2W1.litmus:15]: `ld r0, 1` -> `ld r0, 2`
`ld r0, 2` -> `ld r0, 1`

File: litmus/PTX/Memalloy/WRR+2W1.litmus [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+ st.weak x, r0            | fence.sc.cta             | st.weak x, r0            ;
+                          | ld.weak r2, y            | fence.sc.sys             ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/WRR+2W1.litmus:17]: sys -> gpu .... please be sure this changes is applied to all tests

File: litmus/PTX/Memalloy/WRR+2W2.litmus [hunk 10]
```diff
@@ -0,0 +1,21 @@
+PTX WRR+2W2
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/WRR+2W2.litmus:1]: This test has the same 3 problems as the previous opne

File: litmus/PTX/Memalloy/WRW+2W1.litmus [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+P1:r2=0;
+P2:r0=0;
+P2:r2=0;
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           ;
+ ld r0, 2                 | ld.weak r0, x            | ld r0, 2                 ;
+ st.weak x, r0            | fence.sc.cta             | st.weak x, r0            ;
+                          | ld r2, 1                 | fence.sc.sys             ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/WRW+2W1.litmus:17]: sys => gpu

File: litmus/PTX/Memalloy/WRW+2W2.litmus [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+P1:r0=0;
+P1:r2=0;
+P2:r0=0;
+P2:r2=0;
+x=0;
+y=0;
+}
+ P0@cta 0,gpu 0           | P1@cta 1,gpu 0           | P2@cta 1,gpu 0           ;
+ ld r0, 2                 | ld.weak r0, x            | ld r0, 2                 ;
+ st.weak x, r0            | fence.sc.sys             | st.weak x, r0            ;
```
Comment (hernanponcedeleon) [inline litmus/PTX/Memalloy/WRW+2W2.litmus:16]: sys -> gpu

**Review by hernanponcedeleon — COMMENTED** [review]
Is there any value in keeping the links to memalloy?
If not, please remove them and just keep the second link

File: dartagnan/src/test/resources/PTXv7_5-expected.csv (hunk 1 above)
Comment (tonghaining) [reply dartagnan/src/test/resources/PTXv7_5-expected.csv:226]: added

File: litmus/PTX/Memalloy/IRWIW1.litmus (hunk 2 above)
Comment (tonghaining) [reply litmus/PTX/Memalloy/IRWIW1.litmus]: fixed

File: litmus/PTX/Memalloy/IRWIW2.litmus (hunk 3 above)
Comment (tonghaining) [reply litmus/PTX/Memalloy/IRWIW2.litmus]: fixed

File: litmus/PTX/Memalloy/RWC1.litmus (hunk 4 above)
Comment (tonghaining) [reply litmus/PTX/Memalloy/RWC1.litmus]: Yeah all the `membar.gl` should be `fence.sc.gpu`, fixed now.

File: litmus/PTX/Memalloy/WRR+2W1.litmus (hunk 8 above)
Comment (tonghaining) [reply litmus/PTX/Memalloy/WRR+2W1.litmus]: fixed

**Comment by tonghaining** [comment]
Double checked and we get same result as the alloy tool, tests added in this [branch](https://github.com/tonghaining/mixedproxy/tree/memalloy).

**Review by hernanponcedeleon — APPROVED** [review]

**Pull request merged**